### PR TITLE
[wasm][tests] Fix test failures due to trimming

### PR DESCRIPTION
**Pull request by radical**
- Both EAT, and AOT builds are completely green now.

- Number of tests for EAT goes from 147 -> 185
- Number of tests for AOT goes from 147 -> 174 (the difference is because of aot only failures)

- Test run on a regular wasm build runs 206 libraries

### CODE REVIEW

File: eng/testing/ILDescriptors/ILLink.Descriptors.Serialization.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<linker>
+  <assembly fullname="System.Private.Xml.Linq">
+    <type fullname="System.Xml.Linq.XElement" />
+  </assembly>
+  <assembly fullname="System.Private.DataContractSerialization">
+   <type fullname="System.Runtime.Serialization.KeyValuePairAdapter`2" />
```
Comment (eerhardt) [inline eng/testing/ILDescriptors/ILLink.Descriptors.Serialization.xml:6]: Do we know why this is necessary? This smells like a product bug to me.

cc @joperezr

File: eng/testing/tests.props [hunk 2]
```diff
@@ -6,6 +6,7 @@
     <TestDependsOn Condition="'$(TestNoBuild)' != 'true' and '$(BuildAllProjects)' != 'true'">Build</TestDependsOn>
     <TestDependsOn>$(TestDependsOn);GenerateRunScript;RunTests</TestDependsOn>
     <VSTestNoLogo>true</VSTestNoLogo>
+    <ILDescriptorsPath>$(MSBuildThisFileDirectory)ILDescriptors\</ILDescriptorsPath>
```
Comment (eerhardt) [inline eng/testing/tests.props:9]: I think this should be named `ILLinkDescriptors`. Both the folder name and the MSBuild properties.

File: src/libraries/System.Threading.Tasks.Extensions/tests/ValueTaskTests.cs [hunk 3]
```diff
@@ -1178,6 +1180,7 @@ public void NonGeneric_AsyncMethodBuilderAttribute_ValueTaskAttributed(Type valu
         }
 
         [Theory]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/50722", typeof(PlatformDetection), nameof(PlatformDetection.IsBuiltWithAggressiveTrimming), nameof(PlatformDetection.IsBrowser))]
```
Comment (eerhardt) [inline src/libraries/System.Threading.Tasks.Extensions/tests/ValueTaskTests.cs:1183]: This issue appears to be closed.

File: src/libraries/Common/tests/TestUtilities/TestUtilities.csproj [hunk 4]
```diff
@@ -8,6 +8,7 @@
       and instead use runtime checks.
     -->
     <TargetFrameworks>$(NetCoreAppCurrent);net461</TargetFrameworks>
+    <DefineConstants Condition="'$(EnableAggressiveTrimming)' == 'true'">$(DefineConstants);BUILT_WITH_EAT</DefineConstants>
```
Comment (eerhardt) [inline src/libraries/Common/tests/TestUtilities/TestUtilities.csproj:11]: Let's spell this out. Not everyone understands what `EAT` means.

File: src/libraries/Microsoft.Extensions.Logging/tests/ILLink.Descriptors.xml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<linker>
+  <assembly fullname="System.Diagnostics.TraceSource">
+    <type fullname="System.Diagnostics.TraceListener">
+        <method signature="System.Void .ctor()" />
```
Comment (eerhardt) [inline src/libraries/Microsoft.Extensions.Logging/tests/ILLink.Descriptors.xml:4]: Do we know why this is necessary? This smells like a product bug to me.

File: src/libraries/Microsoft.Extensions.Primitives/tests/Microsoft.Extensions.Primitives.Tests.csproj [hunk 6]
```diff
@@ -8,6 +8,8 @@
   <ItemGroup>
     <PackageReference Include="Moq" Version="$(MoqVersion)" />
     <ProjectReference Include="..\src\Microsoft.Extensions.Primitives.csproj" />
+
+    <TrimmerRootDescriptor Include="$(ILDescriptorsPath)ILLink.Descriptors.Castle.xml" />
```
Comment (eerhardt) [inline src/libraries/Microsoft.Extensions.Primitives/tests/Microsoft.Extensions.Primitives.Tests.csproj:12]: Is there a way to detect if the test has a reference to `Moq`, and add this automatically if it does? That way this doesn't need to be copied everywhere.

File: src/libraries/System.Threading.Tasks.Extensions/tests/ValueTaskTests.cs (hunk 3 above)
Comment (radical) [reply src/libraries/System.Threading.Tasks.Extensions/tests/ValueTaskTests.cs:1183]: Mistakenly closed, reopened now.

File: src/libraries/System.Private.Xml/tests/Readers/CharCheckingReader/System.Xml.RW.CharCheckingReader.Tests.csproj [hunk 7]
```diff
@@ -11,5 +11,7 @@
     <ProjectReference Include="$(CommonTestPath)System\Xml\ModuleCore\ModuleCore.csproj" />
     <ProjectReference Include="$(CommonTestPath)System\Xml\XmlCoreTest\XmlCoreTest.csproj" />
     <ProjectReference Include="..\..\XmlReaderLib\System.Xml.RW.XmlReaderLib.csproj" />
+
+    <TrimmerRootDescriptor Include="$(ILDescriptorsPath)ILLink.Descriptors.ModuleCore.xml" />
```
Comment (eerhardt) [inline src/libraries/System.Private.Xml/tests/Readers/CharCheckingReader/System.Xml.RW.CharCheckingReader.Tests.csproj:15]: It looks like `ModuleCore` is really only used by the Xml tests, and not really "common" across a bunch of tests. Can we just put this in the Xml directory, and use a Directory.Build.props/targets file to include it, so all these files don't need to change?

File: src/libraries/System.Reflection/tests/CoreCLR/System.Reflection.CoreCLR.Tests.csproj [hunk 8]
```diff
@@ -14,5 +14,7 @@
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="..\TestExe\System.Reflection.TestExe.csproj" />
+
+    <TrimmerRootDescriptor Include="$(MSBuildThisFileDirectory)..\ILLink.Descriptors.CoreCLR.xml" />
```
Comment (eerhardt) [inline src/libraries/System.Reflection/tests/CoreCLR/System.Reflection.CoreCLR.Tests.csproj:18]: Since this xml file is only used by this test, can it be moved into this directory?

File: src/libraries/System.Reflection/tests/ILLink.Descriptors.xml [hunk 9]
```diff
@@ -0,0 +1,6 @@
+<linker>
+  <assembly fullname="System.Private.CoreLib">
+    <type fullname="System.Collections.ArrayList" />
```
Comment (eerhardt) [inline src/libraries/System.Reflection/tests/ILLink.Descriptors.xml:3]: Why is this necessary?

File: src/libraries/System.Reflection.MetadataLoadContext/tests/ILLink.Descriptors.xml [hunk 10]
```diff
@@ -0,0 +1,3 @@
+<linker>
+  <assembly fullname="mscorlib" />
```
Comment (eerhardt) [inline src/libraries/System.Reflection.MetadataLoadContext/tests/ILLink.Descriptors.xml:2]: Is this still necessary with the latest linker? It should be preserving facades like this.

Comment (radical) [reply src/libraries/System.Reflection.MetadataLoadContext/tests/ILLink.Descriptors.xml:2]: This was needed with whatever linker is being used with `main` from a couple of days ago.

File: src/libraries/System.Runtime.Extensions/tests/System.Runtime.Extensions.Tests.csproj [hunk 11]
```diff
@@ -92,5 +92,11 @@
     <ProjectReference Include="VoidMainWithExitCodeApp\VoidMainWithExitCodeApp.csproj" />
     <ProjectReference Include="TestApp\TestApp.csproj" />
     <ProjectReference Include="TargetFrameworkNameTestApp\TargetFrameworkNameTestApp.csproj" />
+
+    <TrimmerRootAssembly Include="AssemblyResolveTestApp" />
```
Comment (eerhardt) [inline src/libraries/System.Runtime.Extensions/tests/System.Runtime.Extensions.Tests.csproj:96]: Why do we somtimes use a ILLink.Descriptors.xml file to specify assemblies, but here we use `TrimmerRootAssembly`? Should we be consistent everywhere?

File: src/libraries/System.Text.Json/tests/System.Text.Json.Tests/System.Text.Json.Tests.csproj [hunk 12]
```diff
@@ -4,6 +4,9 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <!-- SYSLIB0020: JsonSerializerOptions.IgnoreNullValues is obsolete -->
     <NoWarn>$(NoWarn);SYSLIB0020</NoWarn>
+
+    <!-- these tests depend on the pdb files -->
```
Comment (eerhardt) [inline src/libraries/System.Text.Json/tests/System.Text.Json.Tests/System.Text.Json.Tests.csproj:8]: Really? Do you know where @layomia @steveharter

File: src/libraries/Common/tests/TestUtilities/TestUtilities.csproj (hunk 4 above)
Comment (radical) [reply src/libraries/Common/tests/TestUtilities/TestUtilities.csproj]: Changed how this is being done now, see [`6d44922` (#51697)](https://github.com/dotnet/runtime/pull/51697/commits/6d44922b92f9df6db107a9b2276f0474df1448fc) and [`d3e215a` (#51697)](https://github.com/dotnet/runtime/pull/51697/commits/d3e215a773af27b628b8213ae1b17228eab90d49)

File: src/libraries/System.Private.Xml/tests/Readers/CharCheckingReader/System.Xml.RW.CharCheckingReader.Tests.csproj (hunk 7 above)
Comment (radical) [reply src/libraries/System.Private.Xml/tests/Readers/CharCheckingReader/System.Xml.RW.CharCheckingReader.Tests.csproj]: 1. moved it to xml tests dir
2. It's used by all the `Reader` tests, so we can use it via `Directory.Build.props` for them.
3. Other directories have more projects though, and it didn't seem worth trying to special case them in a top `Directory.Build.*`

File: src/libraries/Microsoft.Extensions.Logging/tests/ILLink.Descriptors.xml (hunk 5 above)
Comment (radical) [reply src/libraries/Microsoft.Extensions.Logging/tests/ILLink.Descriptors.xml:4]: [`a92333e` (#51697)](https://github.com/dotnet/runtime/pull/51697/commits/a92333e5198daeaf8f73ca9e982dbd4ba064f67e)

```
fail: [FAIL] Microsoft.Extensions.Logging.Test.TraceSourceLoggerTest.Log_Shoud_Add_Exception_To_Message_Whether_Formatter_Is_Null_Or_Not(shouldFormatterBeNull: True)
info: Castle.DynamicProxy.InvalidProxyConstructorArgumentsException : Can not instantiate proxy of class: System.Diagnostics.TraceListener.
info: Could not find a parameterless constructor.
info:    at Castle.DynamicProxy.ProxyGenerator.CreateClassProxyInstance(Type proxyType, List`1 proxyArguments, Type classToProxy, Object[] constructorArguments)
info:    at Castle.DynamicProxy.ProxyGenerator.CreateClassProxy(Type classToProxy, Type[] additionalInterfacesToProxy, ProxyGenerationOptions options, Object[] constructorArguments, IInterceptor[] interceptors)
info:    at Moq.CastleProxyFactory.CreateProxy(Type mockType, IInterceptor interceptor, Type[] interfaces, Object[] arguments)
info:    at Moq.Mock`1[[System.Diagnostics.TraceListener, System.Diagnostics.TraceSource, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]].InitializeInstance()
info:    at Moq.Mock`1[[System.Diagnostics.TraceListener, System.Diagnostics.TraceSource, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]].OnGetObject()
info:    at Moq.Mock.get_Object()
info:    at Moq.Mock`1[[System.Diagnostics.TraceListener, System.Diagnostics.TraceSource, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]].get_Object()
info:    at Microsoft.Extensions.Logging.Test.TraceSourceLoggerTest.<>c__DisplayClass2_0.<Log_Shoud_Add_Exception_To_Message_Whether_Formatter_Is_Null_Or_Not>b__0(ILoggingBuilder builder)
info:    at Microsoft.Extensions.DependencyInjection.LoggingServiceCollectionExtensions.AddLogging(IServiceCollection services, Action`1 configure)
info:    at Microsoft.Extensions.Logging.Test.TestLoggerBuilder.Create(Action`1 configure)
info:    at Microsoft.Extensions.Logging.Test.TraceSourceLoggerTest.Log_Shoud_Add_Exception_To_Message_Whether_Formatter_Is_Null_Or_Not(Boolean shouldFormatterBeNull)
info:    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```

File: src/libraries/System.Private.Xml/tests/Readers/CharCheckingReader/System.Xml.RW.CharCheckingReader.Tests.csproj (hunk 7 above)
Comment (radical) [reply src/libraries/System.Private.Xml/tests/Readers/CharCheckingReader/System.Xml.RW.CharCheckingReader.Tests.csproj]: 1. moved it to xml tests
2. All the `Reader` tests use this, so changed to use `Directory.Build.props` for that
3. Only two other tests are using it, so I let them be in the individual project files

File: src/libraries/System.Reflection/tests/ILLink.Descriptors.xml (hunk 9 above)
Comment (radical) [reply src/libraries/System.Reflection/tests/ILLink.Descriptors.xml:3]: [`e09cad4` (#51697)](https://github.com/dotnet/runtime/pull/51697/commits/e09cad4d691dcb66c837739fc1ca59d84e4226c0)

```
    `IList` implementation on `ArrayList` was getting trimmed

   
      fail: [FAIL] System.Reflection.Tests.TypeInfoTests.IsAssignable(type: typeof(System.Collections.IList), c: typeof(System.Collections.ArrayList), expected: True)
      info: Assert.Equal() Failure
      info: Expected: True
      info: Actual:   False
      info:    at System.Reflection.Tests.TypeInfoTests.IsAssignable(Type type, Type c, Boolean expected) in /Users/radical/dev/r3/src/libraries/System.Reflection/tests/TypeInfoTests.cs:line 580
      info:    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture) in /Users/radical/dev/r3/src/mono/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.cs:line 378
    ```

File: src/libraries/System.Text.Json/tests/System.Text.Json.Tests/System.Text.Json.Tests.csproj (hunk 12 above)
Comment (radical) [reply src/libraries/System.Text.Json/tests/System.Text.Json.Tests/System.Text.Json.Tests.csproj:8]: [`437d614` (#51697)](https://github.com/dotnet/runtime/pull/51697/commits/437d614dc33b98d353eabf50b18f2528e9db1bca)

Debugger tests:
```
    fail: [FAIL] System.Text.Json.Tests.DebuggerTests.DefaultJsonElement
    info: System.NullReferenceException : Object reference not set to an instance of an object.
    info:    at System.Text.Json.Tests.DebuggerTests.GetDebuggerDisplayProperty[JsonElement](JsonElement value)
    info:    at System.Text.Json.Tests.DebuggerTests.DefaultJsonElement()
    info:    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
    fail: [FAIL] System.Text.Json.Tests.DebuggerTests.DefaultUtf8JsonWriter
    info: System.NullReferenceException : Object reference not set to an instance of an object.
    info:    at System.Text.Json.Tests.DebuggerTests.GetDebuggerDisplayProperty[Utf8JsonWriter](Utf8JsonWriter value)
    info:    at System.Text.Json.Tests.DebuggerTests.DefaultUtf8JsonWriter()
    info:    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
    fail: [FAIL] System.Text.Json.Tests.DebuggerTests.DefaultJsonProperty
    info: System.NullReferenceException : Object reference not set to an instance of an object.
    info:    at System.Text.Json.Tests.DebuggerTests.GetDebuggerDisplayProperty[JsonProperty](JsonProperty value)
    info:    at System.Text.Json.Tests.DebuggerTests.DefaultJsonProperty()
    info:    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```

File: eng/testing/ILDescriptors/ILLink.Descriptors.Serialization.xml (hunk 1 above)
Comment (radical) [reply eng/testing/ILDescriptors/ILLink.Descriptors.Serialization.xml]: [`189a79d` (#51697)](https://github.com/dotnet/runtime/pull/51697/commits/189a79d2cdd105914edc29d71d548648f81e9869)

I didn't quite dig into why the linker is missing this.

```
fail: [FAIL] DataContractJsonSerializerTests.DCJS_KeyValuePair

info: System.NullReferenceException : Object reference not set to an instance of an object.
info:    at System.Runtime.Serialization.ReflectionClassWriter.ResolveAdapterType(Object obj, ClassDataContract classContract)
info:    at System.Runtime.Serialization.ReflectionClassWriter.ReflectionWriteClass(XmlWriterDelegator xmlWriter, Object obj, XmlObjectSerializerWriteContext context, ClassDataContract classContract, XmlDictionaryString[] memberNames)
info:    at System.Runtime.Serialization.Json.ReflectionJsonFormatWriter.ReflectionWriteClass(XmlWriterDelegator xmlWriter, Object obj, XmlObjectSerializerWriteContextComplexJson context, ClassDataContract classContract, XmlDictionaryString[] memberNames)
info:    at System.Runtime.Serialization.Json.JsonClassDataContract.WriteJsonValueCore(XmlWriterDelegator jsonWriter, Object obj, XmlObjectSerializerWriteContextComplexJson context, RuntimeTypeHandle declaredTypeHandle)
info:    at System.Runtime.Serialization.Json.JsonDataContract.WriteJsonValue(XmlWriterDelegator jsonWriter, Object obj, XmlObjectSerializerWriteContextComplexJson context, RuntimeTypeHandle declaredTypeHandle)
info:    at System.Runtime.Serialization.Json.DataContractJsonSerializerImpl.WriteJsonValue(JsonDataContract contract, XmlWriterDelegator writer, Object graph, XmlObjectSerializerWriteContextComplexJson context, RuntimeTypeHandle declaredTypeHandle)
info:    at System.Runtime.Serialization.Json.XmlObjectSerializerWriteContextComplexJson.WriteDataContractValue(DataContract dataContract, XmlWriterDelegator xmlWriter, Object obj, RuntimeTypeHandle declaredTypeHandle)
info:    at System.Runtime.Serialization.XmlObjectSerializerWriteContext.SerializeAndVerifyType(DataContract dataContract, XmlWriterDelegator xmlWriter, Object obj, Boolean verifyKnownType, RuntimeTypeHandle declaredTypeHandle, Type declaredType)
info:    at System.Runtime.Serialization.Json.XmlObjectSerializerWriteContextComplexJson.SerializeWithXsiTypeAtTopLevel(DataContract dataContract, XmlWriterDelegator xmlWriter, Object obj, RuntimeTypeHandle originalDeclaredTypeHandle, Type graphType)
info:    at System.Runtime.Serialization.Json.DataContractJsonSerializerImpl.InternalWriteObjectContent(XmlWriterDelegator writer, Object graph)
info:    at System.Runtime.Serialization.Json.DataContractJsonSerializerImpl.InternalWriteObject(XmlWriterDelegator writer, Object graph)
info:    at System.Runtime.Serialization.XmlObjectSerializer.InternalWriteObject(XmlWriterDelegator writer, Object graph, DataContractResolver dataContractResolver)
info:    at System.Runtime.Serialization.XmlObjectSerializer.WriteObjectHandleExceptions(XmlWriterDelegator writer, Object graph, DataContractResolver dataContractResolver)
info:    at System.Runtime.Serialization.XmlObjectSerializer.WriteObjectHandleExceptions(XmlWriterDelegator writer, Object graph)
info:    at System.Runtime.Serialization.Json.DataContractJsonSerializerImpl.WriteObject(XmlDictionaryWriter writer, Object graph)
info:    at System.Runtime.Serialization.Json.DataContractJsonSerializerImpl.WriteObject(Stream stream, Object graph)
info:    at System.Runtime.Serialization.Json.DataContractJsonSerializer.WriteObject(Stream stream, Object graph)
info:    at DataContractJsonSerializerTests.SerializeAndDeserialize[KeyValuePair`2](KeyValuePair`2 value, String baseline, DataContractJsonSerializerSettings settings, Func`1 serializerFactory, Boolean skipStringCompare)
info:    at DataContractJsonSerializerTests.DCJS_KeyValuePair()
info:    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```

File: src/libraries/Microsoft.Extensions.Primitives/tests/Microsoft.Extensions.Primitives.Tests.csproj (hunk 6 above)
Comment (radical) [reply src/libraries/Microsoft.Extensions.Primitives/tests/Microsoft.Extensions.Primitives.Tests.csproj]: We could look for `Moq` in `@(PackageReference)`. I tried to find all the references to `Moq`, and `*Castle.xml`:

```
$ git grep -E "PackageReference.*Moq|Castle.xml"

src/libraries/Microsoft.Extensions.Configuration.FileExtensions/tests/Microsoft.Extensions.Configuration.FileExtensions.Tests.csproj:    <TrimmerRootDescriptor Include="$(ILLinkDescriptorsPath)ILLink.Descriptors.Castle.xml" />
src/libraries/Microsoft.Extensions.Configuration.FileExtensions/tests/Microsoft.Extensions.Configuration.FileExtensions.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/Microsoft.Extensions.Configuration/tests/Microsoft.Extensions.Configuration.Tests.csproj:    <TrimmerRootDescriptor Include="$(ILLinkDescriptorsPath)ILLink.Descriptors.Castle.xml" />
src/libraries/Microsoft.Extensions.Configuration/tests/Microsoft.Extensions.Configuration.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/Microsoft.Extensions.DependencyModel/tests/Microsoft.Extensions.DependencyModel.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/Microsoft.Extensions.FileProviders.Composite/tests/Microsoft.Extensions.FileProviders.Composite.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/Microsoft.Extensions.FileProviders.Composite/tests/Microsoft.Extensions.FileProviders.Composite.Tests.csproj:    <TrimmerRootDescriptor Include="$(ILLinkDescriptorsPath)ILLink.Descriptors.Castle.xml" />
src/libraries/Microsoft.Extensions.FileProviders.Physical/tests/Microsoft.Extensions.FileProviders.Physical.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/Microsoft.Extensions.Hosting.Unit.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/Microsoft.Extensions.Http/tests/Microsoft.Extensions.Http.Tests/Microsoft.Extensions.Http.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/Microsoft.Extensions.Http/tests/Microsoft.Extensions.Http.Tests/Microsoft.Extensions.Http.Tests.csproj:    <TrimmerRootDescriptor Include="$(ILLinkDescriptorsPath)ILLink.Descriptors.Castle.xml" />
src/libraries/Microsoft.Extensions.Logging/tests/Common/Microsoft.Extensions.Logging.Tests.csproj:    <TrimmerRootDescriptor Include="$(ILLinkDescriptorsPath)ILLink.Descriptors.Castle.xml" />
src/libraries/Microsoft.Extensions.Logging/tests/Common/Microsoft.Extensions.Logging.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/Microsoft.Extensions.Primitives/tests/Microsoft.Extensions.Primitives.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/Microsoft.Extensions.Primitives/tests/Microsoft.Extensions.Primitives.Tests.csproj:    <TrimmerRootDescriptor Include="$(ILLinkDescriptorsPath)ILLink.Descriptors.Castle.xml" />
src/libraries/System.ComponentModel.TypeConverter/tests/System.ComponentModel.TypeConverter.Tests.csproj:    <TrimmerRootDescriptor Include="$(ILLinkDescriptorsPath)ILLink.Descriptors.Castle.xml" />
src/libraries/System.ComponentModel.TypeConverter/tests/System.ComponentModel.TypeConverter.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/System.Runtime/tests/System.Runtime.Tests.csproj:    <TrimmerRootDescriptor Include="$(ILLinkDescriptorsPath)ILLink.Descriptors.Castle.xml" />
src/libraries/System.Runtime/tests/System.Runtime.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/System.Text.Encoding/tests/System.Text.Encoding.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
```

.. and there seem to be some projects that either don't need the linker xml, or maybe I didn't get to failing test because of `Moq` yet.

Should we add that irrespective? or maybe doing it as-needed per project is better?

**Comment by radical** [comment]
Both EAT, and AOT builds are green now, all the (enabled) tests passing.

@lewing @steveisok @SamMonoRT

**Review by lewing — APPROVED** [review]
This looks good to me.  Can we do an immediate follow up with xml comments that point to the issue/reason the changes were added?

File: src/libraries/System.Text.Json/tests/System.Text.Json.Tests/System.Text.Json.Tests.csproj (hunk 12 above)
Comment (eerhardt) [reply src/libraries/System.Text.Json/tests/System.Text.Json.Tests/System.Text.Json.Tests.csproj:8]: This one could be fixed in the test pretty easily - 

```C#
        private static string GetDebuggerDisplayProperty<[DynamicallyAccessedMembers.PublicProperties] T>(T value)
        {
            return (string)typeof(T).GetProperty("DebuggerDisplay", BindingFlags.Instance | BindingFlags.NonPublic).GetValue(value);
        }
```

File: eng/testing/ILDescriptors/ILLink.Descriptors.Serialization.xml (hunk 1 above)
Comment (eerhardt) [reply eng/testing/ILDescriptors/ILLink.Descriptors.Serialization.xml]: Opened https://github.com/dotnet/runtime/issues/52083 for this.

File: src/libraries/System.Reflection/tests/ILLink.Descriptors.xml (hunk 9 above)
Comment (eerhardt) [reply src/libraries/System.Reflection/tests/ILLink.Descriptors.xml:3]: If it is just that one test, maybe adding a `DynamicDependency` on the test would be a better approach. Then it localizes it, and you can tell why it is needed. Having it rooted here makes it hard to reason about.

File: src/libraries/Microsoft.Extensions.Logging/tests/ILLink.Descriptors.xml (hunk 5 above)
Comment (eerhardt) [reply src/libraries/Microsoft.Extensions.Logging/tests/ILLink.Descriptors.xml:4]: I see, we are using `Mock<TraceListener>` in the test. Maybe the test could be annotated with `DynamicDependency` instead, so it is more discoverable about why this needs to be preserved.
